rope_.d

```diff
@@ -74,19 +74,19 @@ template Rope(T,S=void)if(is(S==void) || is(typeof((S a,S b)=>a.combine(b)))){
 		static if(segtree){
 			@property value(){ return isArray()?tree?tree.value:S.init:rope.value; }
 			private this(T[] array, SliceSegTree* tree){
-				if(array==[]) return;
 				this.array = array;
+				if(array==[]) return;
 				this.tree = tree;
 			}
 		}
 
 		private this(RopeImpl* rope){ this.rope = rope; }
 		invariant(){ assert(cast(void*)rope is array.ptr); }
-		private union{
+		/*private*/ union{
```

Comment:
Ahh, makes sense now ;-)

I've renamed the member to `array_`. After that I hit another access violation by `Variant.toHash` not being used as the custom hash function. With an (ugly) fix to ths, ttuple.d can be processed.